src/node_sea.cc

```diff
@@ -73,6 +90,71 @@ std::tuple<int, char**> FixupArgsForSEA(int argc, char** argv) {
   return {argc, argv};
 }
 
+ExitCode BuildSingleExecutableBlob(const std::string& config_path) {
+  std::string config;
+  int r = ReadFileSync(&config, config_path.c_str());
+  if (r != 0) {
+    fprintf(stderr,
+            "Cannot read single executable configuration from %s\n",
+            config_path.c_str());
+    return ExitCode::kGenericUserError;
+  }
+
+  std::string main_path;
+  std::string output_path;
+  {
+    JSONParser parser;
+    if (!parser.Parse(config)) {
+      fprintf(stderr, "Cannot parse JSON from %s\n", config_path.c_str());
+      return ExitCode::kGenericUserError;
+    }
+
+    main_path = parser.GetTopLevelField("main").value_or(std::string());
+    if (main_path.empty()) {
+      fprintf(stderr,
+              "\"main\" field of %s is not a string\n",
+              config_path.c_str());
+      return ExitCode::kGenericUserError;
+    }
+
+    output_path = parser.GetTopLevelField("output").value_or(std::string());
+    if (output_path.empty()) {
+      fprintf(stderr,
+              "\"output\" field of %s is not a string\n",
+              config_path.c_str());
+      return ExitCode::kGenericUserError;
+    }
+  }
```

Comment:
WDYT about creating a separate private method for this job? (`ParseSeaConfig`) ?

Reply:
And we could follow the same logic for writing the output file below.